src/plugins/controls/public/control_types/options_list/options_list_editor.tsx

```diff
@@ -40,20 +37,21 @@ export const OptionsListEditor = ({
   setDefaultTitle,
   getRelevantDataViewId,
   setLastUsedDataViewId,
+  selectedField,
+  setSelectedField,
 }: ControlEditorProps<OptionsListEmbeddableInput>) => {
   // Controls Services Context
   const { dataViews } = pluginServices.getHooks();
   const { getIdsWithTitle, getDefaultId, get } = dataViews.useService();
 
   const [state, setState] = useState<OptionsListEditorState>({
-    fieldName: initialInput?.fieldName,
     singleSelect: initialInput?.singleSelect,
     dataViewListItems: [],
   });
 
   useMount(() => {
     let mounted = true;
-    if (state.fieldName) setDefaultTitle(state.fieldName);
+    if (selectedField) setDefaultTitle(selectedField);
```

Comment:
I needed to be able to **unset** the selected field from the general `control_editor` rather than having it be entirely handled for each type-specific editor - otherwise, when the user selects a different type, the previous field would stay selected so, when saved, the control would be broken because the old field type was incompatible with the new control type. 

So, I decided to manage this state from `control_editor.tsx` and simply passing in the value and setter in to each type-specific editor. Now, when the user selects a different control type on edit, the save button should be disabled until they select a new field.